src/c++/perf_analyzer/genai-perf/genai_perf/parser.py

```diff
@@ -679,45 +680,75 @@ def compare_handler(args: argparse.Namespace):
     plot_manager.generate_plots()
 
 
-### Entrypoint ###
+def profile_handler(args, extra_args):
+    from genai_perf.wrapper import Profiler
 
+    Profiler.run(args=args, extra_args=extra_args)
+
+
+### Parser Initialization ###
 
-def parse_args():
-    argv = sys.argv
 
+def init_parsers():
     parser = argparse.ArgumentParser(
         prog="genai-perf",
         description="CLI to profile LLMs and Generative AI models with Perf Analyzer",
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
     )
-    parser.set_defaults(func=profile_handler)
-
-    # Conceptually group args for easier visualization
-    _add_endpoint_args(parser)
-    _add_input_args(parser)
-    _add_profile_args(parser)
-    _add_output_args(parser)
-    _add_other_args(parser)
+    parser.add_argument(
+        "--version",
+        action="version",
+        version="%(prog)s " + __version__,
+        help=f"An option to print the version and exit.",
+    )
 
     # Add subcommands
     subparsers = parser.add_subparsers(
         help="List of subparser commands.", dest="subcommand"
     )
-    compare_parser = _parse_compare_args(subparsers)
+    _ = _parse_compare_args(subparsers)
+    _ = _parse_profile_args(subparsers)
+    subparsers.required = True
 
-    # Check for passthrough args
+    return parser
+
+
+def get_passthrough_args_index(argv: list) -> int:
     if "--" in argv:
         passthrough_index = argv.index("--")
         logger.info(f"Detected passthrough args: {argv[passthrough_index + 1:]}")
     else:
         passthrough_index = len(argv)
 
+    return passthrough_index
+
+
+def refine_args(
+    parser: argparse.ArgumentParser, args: argparse.Namespace
+) -> argparse.Namespace:
+    if args.subcommand == "profile":
```

Comment:
should we create an enum for subcommands?

Reply:
Good idea! Done.